### PR TITLE
Coinfest ERC20 Solidity Verification Example

### PR DESCRIPTION
**Pull request by ehildenb**

### CODE REVIEW

File: tests/specs/examples/erc20-spec.md [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+       requires TOTALSUPPLY_KEY ==Int #hashedLocation("Solidity", 2, .IntList)
+        andBool TOTALSUPPLY     ==Int #lookup(ACCT_STORAGE,  TOTALSUPPLY_KEY)
+```
+
+### Calling Approve works
+
+-   Everything from `<mode>` to `<substate>` is boilerplate.
+-   We are setting `<callData>` to `approve(SPENDER, AMOUNT)`.
+-   We ask the prover to show that in all cases, we will end in `EVMC_SUCCESS` (rollback) when SENDER or OWNER is not address(0), and that we will end in `EVMC_REVERT` when either one of them is.
```
Comment (denis-bogdanas) [inline tests/specs/examples/erc20-spec.md:249]: I guess `EVMC_REVERT` (rollback)

**Review by denis-bogdanas — COMMENTED** [review]
One issue in comment, the rest looks good.

File: tests/specs/examples/erc20-spec.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+        return true;
+    }
+
+    function _transfer(address from, address to, uint256 amount) internal {
+        require(from != address(0), "ERC20: transfer from the zero address");
+        require(to != address(0), "ERC20: transfer to the zero address");
+        uint256 toBalance   = _balances[to];
+        uint256 fromBalance = _balances[from];
+        require(fromBalance >= amount, "ERC20: transfer amount exceeds balance");
+        uint256 fromBalanceUpdated = fromBalance - amount;
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply tests/specs/examples/erc20-spec.md:83]: Yeah it's intended. This is buggy code that was supplied as part of a demo, and we were only able to get through proving the getters (decimals and totalSupply) and the approve function in the time allocated for the workshop.

File: tests/specs/examples/erc20-spec.md (hunk 1 above)
Comment (ehildenb) [reply tests/specs/examples/erc20-spec.md:249]: Good catch.

Comment (ehildenb) [inline tests/specs/examples/erc20-spec.md:249]:
```suggestion
-   We ask the prover to show that in all cases, we will end in `EVMC_SUCCESS` when `SENDER` or `OWNER` is not `address(0)`, and that we will end in `EVMC_REVERT` (rollback) when either one of them is.
```

**Review by denis-bogdanas — APPROVED** [review]

**Pull request merged**